test/functional/eval/timer_spec.lua

```diff
@@ -19,72 +19,89 @@ describe('timers', function()
   end)
 
   it('works one-shot', function()
-    command("call timer_start(50, 'MyHandler')")
-    eq(0,eval("g:val"))
-    run(nil, nil, nil, load_adjust(200))
-    eq(1,eval("g:val"))
+    eq(0, eval("[timer_start(10, 'MyHandler'), g:val][1]"))
+    run(nil, nil, nil, load_adjust(10))
+    retry(nil, load_adjust(200), function()
+      eq(1, eval("g:val"))
+    end)
```

Comment:
retry() here seems weird. we want to test that the timer did not run again, right? so retry(200) is very different than the old run(200)

Reply:
Good point, adjusted.